### ISSUE TITLE
Decimals for Viral Loads Results in the 1.6 xsd

### ISSUE THREAD

**Issue by brightoibe** [nmrs-nigeria/openmrs-module-nigeriaemr#130]
Hello Team,

Is it Ok for us to edit the xsd to allow all numeric lab tests to accept decimals? I believe this will give us more flexibility than forcing all of them to be casted to integers.

**Comment by wandechris** [comment]
@brightoibe no, you can't edit it. the Xsd is the specification from NDR because that's what NDR is expecting. all changes will need to come from them.

**Comment by wandechris** [comment]
@ahmedsylvanus @caniekwe is this something that needs to be prioritized?

**Comment by caniekwe** [comment]
I agree with you @wandechris. The XSD should not be edited. The viral load is an integer value and not decimal. or Do you have decimal viral load results in your EMR @brightoibe

**Comment by brightoibe** [comment]
@wandechris and Team. I agree this may not be a change we can make now because other non-CDC partners are using the same schema. My sugestions was that since "ValueNumeric" is used to hold both integers lab tests and float lab tests, it will be nice if we make it a decimal type because a decimal type can contain floats. There are a number of lab tests that can have floats in our lab forms and all of them are captured using the same field "ValueNumeric".

**Comment by talk2morris** [comment]
@brightoibe I understand you point even though any change to the NDR XSD needs to go through proper documentation before any change gets in, in addition to what you observe, I would propose a string type on NMRS and NDR and the reason is because from my experience with LIMS the testing machine can sometimes produce results such as >10000000, <20, 'sample not detected' as valid results and these are not numeric.

**Comment by wonderfulguru1** [comment]
I will want to agree with @talk2morris  on this. but the issue accepting string is that these results are used for calculations like Eligibility, Viral load suppression  e.tc, so how would you calculate with >10000000 or  <20.

**Comment by brightoibe** [comment]
Hello Team. How about making striping the `>` and `<`  from the string and converting then to decimals, so we can still use it for different kinds of analysis. E.g if your viral load is "<20" then it should be stored as 20, as long as the health program is concerned, anything 20 and below is undetected.

**Comment by wandechris** [comment]
@brightoibe that will be on the NDR level, NDR might want to account for those extras, it's not right for the plugin to truncate it

**Comment by caniekwe** [comment]
My suggestion here is to let NDR worry about NDR issues. Whatever standard NDR sets for us, we follow. If the standard set by the NDR does not work for us, then we raise it with the NDR and not from the EMR

**Comment by brightoibe** [comment]
Good evening @caniekwe @wandechris and Team. I just got a report from site today that they saw Viral Loads with decimals. These decimals usually reduce concurrence rate when they M&E teams compare NDR and EMR. I think we need to talk to a Lab person on this. Thank you
